### ISSUE TITLE
namespace_linter() is OS-dependent

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#3075]
Consider in {knitr}:

https://github.com/yihui/knitr/blob/646ba86a2f2220454265b4ff1038fecfc5381923/R/plot.R#L70

```r
win.metafile = grDevices::win.metafile
```

{grDevices} _does_ export `win.metafile`, but _only_ when compiled on Windows:

https://github.com/r-devel/r-svn/blob/bb087814dacb62ed5f44a3c449c31268e137b710/src/library/grDevices/NAMESPACE#L76-L86

It's not ideal for the output of `namespace_linter()` to be platform-dependent, e.g. it flags this when run on Unix:

```r
lint("R/plot.R", namespace_linter())
# R/plot.R:70:31: warning: [namespace_linter] 'win.metafile' is not exported from {grDevices}.
#     win.metafile = grDevices::win.metafile,
#                               ^~~~~~~~~~~~
```
